content/blog/angular-extend-class/index.md

```diff
@@ -0,0 +1,1010 @@
+
+Further, there's a bit of a code smell present by manually calling these functions.
+
+Fortunately, there's a better way...
+
+## The Angular way to fix the code
+
+While mutable properties can get the job done, they're far from optimal. Let's instead leverage `rxjs`, which is built into Angular after all, to create an observable.
+
+For this type of DOM event listening, RxJS exposes a [`fromEvent`](https://rxjs.dev/api/index/function/fromEvent) method that we can pipe into a [`map`](https://rxjs.dev/api/operators/map) to create an [Observable](https://rxjs.dev/guide/observable).
+
+```typescript
+import {fromEvent, debounceTime, map, Subject, takeUntil, Observable} from 'rxjs';
+
+interface WindowSize {
+  readonly height: number;
+  readonly width: number;
+}
+
+@Injectable()
+class WindowSizeService {
+  private destroy$ = new Subject<void>();
+
+  size$: Observable<WindowSize>;
+
+  constructor(@Inject(DOCUMENT) document: Document) {
+    const window = document.defaultView!;
+    this.size$ = fromEvent(window, 'resize').pipe(
+      debounceTime(50),
+      map(() => ({
+        height: window.innerHeight,
+        width: window.innerWidth,
+      })),
+      takeUntil(this.destroy$),
+      startWith({
+        height: window.innerHeight,
+        width: window.innerWidth,
+      })
+    );
+  }
+
+  cleanup() {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+}
+```
+
+While we still need a `cleanup` method, this is a much more straightforward setup process that's much more Angular-ific!
+
+As an added benifit, we now can utilize an [`AsyncPipe`](https://angular.io/api/common/AsyncPipe) in order to listen for changes on the `size$` observable:
+
+```typescript
+@Component({
+  selector: 'app-root',
+  template: `
+    <p *ngIf="windowSize.size$ | async as size">The window is {{size.height}}px high and {{size.width}}px wide</p>
+  `,
+  providers: [WindowSizeService]
+})
+class AppComponent implements OnDestroy {
+  windowSize = inject(WindowSizeService);
+
+  ngOnDestroy() {
+    this.windowSize.cleanup();
+  }
+}
+```
+
+
+
+# Conclusion
+
+And that's it! I hope this has been an insightful look into how you can extend component logic.
+
+And this isn't an area of stagnation within Angular - they're introducing new functionality to share component logic using [the upcoming `hostDirectives` API](https://github.com/angular/angular/pull/46868).
+
+Hey, while you're here - do you want to learn more Angular in-depth like this? Maybe you've been working in Angular for some time and want to learn React or Vue, but not start from scratch?
+
+Check out [my free book, "The Framework Field Guide", that teaches React, Angular, and Vue all at the same time.](https://framework.guide).
+
+> A huge huge thank you to [Lars Gyrup Brink Nielsen](https://twitter.com/LayZeeDK) for helping revise this article, and providing some of the methodologies and code samples present throughout, including the wonderful RxJS code sample.
```

Comment:
```suggestion
```

Seems a bit odd to attribute a co-author as-if they weren't a co-author, so I'm removing this. @LayZeeDK put just as much effort into this article as I had! ❤️